src/remote_prune

```diff
@@ -0,0 +1,236 @@
+    parser.add_argument("--aws-credentials",help="AWS Credentials",
+                        default=os.environ.get("GCP_PROJECT_NAME"))
+
+    # subparsers = parser.add_subparsers(dest='cmd', title='subcommands')
+    # subparsers.required = True
+
+    # s3 = subparsers.add_parser('s3', help='Prune s3 buckets')
+    # s3.add_argument("--bucket", help="Bucket name")
+    # s3.add_argument("--prefix", help="Key prefix")
+    # s3.add_argument("--force", help="Wipe s3 key ignoring the errors",
+    #                 action='store_true')
+
+    args = parser.parse_args()
+
+    cloud_config = {
+        'azure': {
+            'credentials': args.azure_credentials,
+            'resource-group': args.azure_resource_group,
+        },
+        'gcp': {
+            'json-key': args.gcp_json_key,
+            'project': args.gcp_project,
+        }
+    }
+
+    policy = open(f"{args.policy}")
+    policy = yaml.safe_load(policy)
+    stream = args.stream
+
+    if stream in policy:
+        # If the build key is set in the policy file, then the cloud-uploads key must
+        # also be present, and the duration of cloud-uploads must be equal or shorter
+        if "build" in policy[stream].keys():
+            cloudUploadsCheck(policy[stream])
+        buildJsonData = getBuildsForStream(stream)
+        # action is basically whatever is needed to be pruned for the respective stream
+        for action in policy[stream]:
+            years = policy[stream][action]
+            #Ref date: Artifacts older than this date will be deleted
+            refDate = datetime.datetime.now() - relativedelta(years=int(years[:-1]))
+            print(f"Reference Date for action {action} is {refDate}")
+
+            for index, build in enumerate(buildJsonData["builds"]):
+                build_id = build["id"]
+                if "policy-cleanup" in build.keys():
+                    if action in build["policy-cleanup"]:
+                        print(f"The {build_id} has already had {action} pruning completed")
+                        break
+                timestamp = build_id.split('.')[1]
+                buildDate = datetime.datetime(int(timestamp[0:4]), int(timestamp[4:6]), int(timestamp[-2:]))
+                if buildDate < refDate:
+                    # For testing pusposes only using one build
+                    if build_id == "35.20210508.91.0":
+                        for arch in build["arches"]:
+                            print(f"Pruning {years} old {action} for {build_id} for {arch} in {stream}")
+                            # buildFetch(args.stream, build_id, arch)
+                            metaJson = open(f"builds/{build_id}/{arch}/meta.json")
+                            metaData = json.load(metaJson)
+
+                            images = {
+                                "amis": metaData.get("amis") or [],
+                                "azure": metaData.get("azure") or [],
```

Comment:
for FCOS at least we don't have any cloud resources for Azure that we need to cleanup (yet).

For RHCOS I think we do have some object storage uploads we do.